### PR TITLE
JFrog Pipeline changes

### PR DESCRIPTION
**Pull request by danielmkn**

### CODE REVIEW

File: TFproviderTest.yml [hunk 1]
```diff
@@ -171,29 +171,30 @@ pipelines:
             - cat GNUmakefile | sed -e "s/darwin_amd64/linux_amd64/g" > GNUmakefile.tmp
             - cp GNUmakefile.tmp GNUmakefile && rm GNUmakefile.tmp
             - cat GNUmakefile
-            - cat sample.tf | sed -e "s/version =.*/version = \"${NEXT_PROVIDER_VERSION}\"/g" > sample.tf.tmp
-            - cp sample.tf.tmp sample.tf && rm sample.tf.tmp
-            - cat sample.tf
+            ## commented out before the first version released. Version is hardcoded to 0.0.1
```
Comment (alexhung) [inline TFproviderTest.yml:174]: If we aren't updating the version in `sample.tf` then please delete these lines.

File: TFproviderTest.yml [hunk 2]
```diff
@@ -171,29 +171,30 @@ pipelines:
...
+            #- cat sample.tf
             - echo "Add variables needed to run Terraform Provider Xray"
-            - export ARTIFACTORY_URL=${jdp_url} && echo ${ARTIFACTORY_URL}
+            - export JFROG_URL=${jdp_url} && echo ${JFROG_URL}
             - export ARTIFACTORY_USERNAME=admin
             - export ARTIFACTORY_PASSWORD=${NEW_RT_PASSWORD} && echo ${ARTIFACTORY_PASSWORD}
             - echo "Get cookie to generate Access token"
             - >-
-              export COOKIES=$(curl -c - "${ARTIFACTORY_URL}/ui/api/v1/ui/auth/login?_spring_security_remember_me=false" \
+              export COOKIES=$(curl -c - "${JFROG_URL}/ui/api/v1/ui/auth/login?_spring_security_remember_me=false" \
```
Comment (alexhung) [inline TFproviderTest.yml:184]: Let's put a comment here explaining why we use the UI API to create the token (from cookie).

File: TFproviderTest.yml [hunk 3]
```diff
@@ -171,29 +171,30 @@ pipelines:
...
                             --header "accept: application/json, text/plain, */*" \
                             --header "content-type: application/json;charset=UTF-8" \
                             --header "x-requested-with: XMLHttpRequest" \
                             -d '{"user":"admin","password":"'"${ARTIFACTORY_PASSWORD}"'","type":"login"}' | grep FALSE)
```
Comment (alexhung) [inline TFproviderTest.yml:188]: Why aren't we using the env var `ARTFIACTORY_USERNAME` here?

File: TFproviderTest.yml [hunk 4]
```diff
@@ -226,15 +227,18 @@ pipelines:
             - export PR_URL=${res_GitHubTFProviderXrayRepoJFrog_commitUrl}
             - export PR_COMMITTER=${res_GitHubTFProviderXrayRepoJFrog_committerLogin}
             - export PR_BRANCH=${res_GitHubTFProviderXrayRepoJFrog_headCommitRef}
-            - export PR_TITLE=${res_GitHubTFProviderXrayRepoJFrog_commitMessage}
+            - export PR_TITLE=$(echo ${res_GitHubTFProviderXrayRepoJFrog_commitMessage} | sed -e 's/\&/\&amp;/g' -e 's/>/\&gt;/g' -e 's/</\&lt;/g')
```
Comment (alexhung) [inline TFproviderTest.yml:230]: Please add comment here on why we escaping three characters

File: TFproviderTest.yml [hunk 5]
```diff
@@ -171,29 +171,30 @@ pipelines:
...
             - export ACCESSTOKEN=$(echo $COOKIES | grep ACCESSTOKEN | awk '{print $14 }') # awk returns null on Mac, and the actual key on Ubuntu
             - >-
-              export ACCESS_KEY=$(curl -g --request GET "${ARTIFACTORY_URL}/ui/api/v1/system/security/token?services[]=all" \
+              export ACCESS_KEY=$(curl -g --request GET "${JFROG_URL}/ui/api/v1/system/security/token?services[]=all" \
                                   --header "accept: application/json, text/plain, */*" \
                                   --header "x-requested-with: XMLHttpRequest" \
                                   --header "cookie: ACCESSTOKEN=${ACCESSTOKEN}; REFRESHTOKEN=${REFRESHTOKEN}")
-            - add_run_variables JFROG_ACCESS_TOKEN=${ACCESS_KEY}
-            - echo "Unset ARTIFACTORY_USERNAME and ARTIFACTORY_PASSWORD, acceptance test will use JFROG_ACCESS_TOKEN instead"
+            - export XRAY_ACCESS_TOKEN=${ACCESS_KEY}
```
Comment (alexhung) [inline TFproviderTest.yml:196]: I'd rather continue using `JFROG_ACCESS_TOKEN` to be consistent. But if pipeline doesn't like it, not much we can do.

File: TFproviderTest.yml (hunk 1 above)
Comment (alexhung) [reply TFproviderTest.yml:174]: Same with L165...

File: TFproviderTest.yml [hunk 6]
```diff
@@ -171,29 +171,30 @@ pipelines:
             - cat GNUmakefile | sed -e "s/darwin_amd64/linux_amd64/g" > GNUmakefile.tmp
```
Comment (alexhung) [inline TFproviderTest.yml:171]: If we add

```
TARGE_ARCH?=darwin_amd64
```
to `GNUmakefile`

And replace all instances of `darwin_amd64` with `${TARGE_ARCH}`

Then we won't need to do these sed and file gymnastic.

Just `make install -e TARGET_ARCH=linux_amd64` will work.

**Review by alexhung — CHANGES_REQUESTED** [review]

File: TFproviderTest.yml (hunk 1 above)
Comment (danielmkn) [reply TFproviderTest.yml:174]: we will update it with the first version will be released

File: TFproviderTest.yml (hunk 2 above)
Comment (danielmkn) [reply TFproviderTest.yml:184]: Agree, added. I'm echoing it, so we can see it in the log as well

File: TFproviderTest.yml (hunk 1 above)
Comment (danielmkn) [reply TFproviderTest.yml:174]: removing it, not it's done in the makefile, like in Projects and RT providers

File: TFproviderTest.yml (hunk 3 above)
Comment (danielmkn) [reply TFproviderTest.yml:188]: replaced. Admin is a default user, so I declared that var just as a reminder I guess..

File: TFproviderTest.yml (hunk 4 above)
Comment (danielmkn) [reply TFproviderTest.yml:230]: done

File: TFproviderTest.yml (hunk 6 above)
Comment (danielmkn) [reply TFproviderTest.yml]: done

**Review by alexhung — APPROVED** [review]
Let's test this in Pipeline!

**Pull request merged**